pypush/apns/lifecycle.py

```diff
@@ -6,28 +6,35 @@
 import time
 import typing
 from contextlib import asynccontextmanager
+from hashlib import sha1
 
 import anyio
 from anyio.abc import TaskGroup
 from cryptography import x509
 from cryptography.hazmat.primitives import hashes, serialization
 from cryptography.hazmat.primitives.asymmetric import padding, rsa
 
-from . import protocol, transport, _util
+from . import _util, filters, protocol, transport
 
 
 @asynccontextmanager
 async def create_apns_connection(
     certificate: x509.Certificate,
     private_key: rsa.RSAPrivateKey,
     token: typing.Optional[bytes] = None,
+    sandbox: bool = False,
     courier: typing.Optional[str] = None,
 ):
     async with anyio.create_task_group() as tg:
-        conn = Connection(tg, certificate, private_key, token, courier)
+        conn = Connection(
+            tg, certificate, private_key, token, sandbox, courier
+        )  # Await connected for first time here, so that base token is set
+        await conn._connected.wait()
```

Comment:
This is kind of sketchy, probably good for people to be able to instantiate a `Connection` manually (if they have their own `TaskGroup`), but they wouldn't know to wait on `_connected` (especially due to it's implementation-detail nature).

Really the only reason it is necessary is for the first connection, so that you don't accidentally try to do anything before `base_token` is set, if you already have a token then things should just wait for the connection to be established patiently.